org.eclipse.wildwebdeveloper/src/org/eclipse/wildwebdeveloper/debug/AbstractDebugAdapterLaunchShortcut.java

```diff
@@ -154,7 +156,7 @@ public void launch(IEditorPart editor, String mode) {
 
 	private void launch(String mode, ILaunchConfiguration[] configurations) {
 		if (configurations.length == 1) {
-			CompletableFuture.runAsync(() -> DebugUITools.launch(configurations[0], mode));
+			DebugUITools.launch(configurations[0], mode);
```

Comment:
IIRC, it was important that this method would be non-blocking and that's why it got wrapped in a Future.
I think calling `Display.getDefault().asyncExec(() -> DebugUITools.launch(configurations[0], mode))` is safer here.

Reply:
Done.